File: archive/max-consecutive-ones.js
const findMaxConsecutiveOnes = function(nums) {
  let max = 0
  let mem = 0

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] === 1) {
      mem++
    } else {
      mem = 0
    }

    max = Math.max(mem, max)
  }

  return max
}

console.log(findMaxConsecutiveOnes([1,0,1,1,0,1]))

File: max-consecutive-ones.go
func findMaxConsecutiveOnes(nums []int) int {
    c:= 0; max:=0

    for i:=0; i < len(nums); i++ {
        if nums[i] == 0 {c = 0}
        if nums[i] == 1 {c++}

        if (c > max) {max = c}
    }

    return max
}